pipelines/services/run.go

```diff
@@ -0,0 +1,157 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type RunService struct {
+	client *jfroghttpclient.JfrogHttpClient
+	auth.ServiceDetails
+}
+
+func NewRunService(client *jfroghttpclient.JfrogHttpClient) *RunService {
+	return &RunService{client: client}
+}
+
+const (
+	runStatus            = "api/v1/search/pipelines/"
+	triggerpipeline      = "api/v1/pipelines/trigger"
+	pipelineSyncStatus   = "api/v1/pipelineSyncStatuses"
+	pipelineResources    = "api/v1/pipelineSources"
+	cancelRunPath        = "api/v1/runs/:runId/cancel"
+	syncPipelineResource = "api/v1/pipelineSources"
+	resourceVersions     = "api/v1/resourceVersions"
+)
+
+func (rs *RunService) GetRunStatus(branch, pipeName string, isMultiBranch bool) (*PipelineRunStatusResponse, error) {
+	httpDetails := rs.getHttpDetails()
+
+	// query params
+	queryParams := make(map[string]string, 0)
+	if isMultiBranch { // add this query param only when pipeline source is multi-branch
+		queryParams["pipelineSourceBranch"] = branch
+	}
+	if pipeName != "" {
+		queryParams["name"] = pipeName
+	}
+
+	// URL Construction
+	uri, pipeURLErr := constructPipelinesURL(queryParams, rs.ServiceDetails.GetUrl(), runStatus)
```

Comment:
pipeURLErr --> err

Let's use `err` when possible.

Reply:
There are 2 reasons why I named instead of `err`
1. To add more info to the error variable name while returning and handling errors.
2. To avoid issues caused by reusing Go error variables

Reply:
Are you sure you cannot reuse the name 'err' in this case?

Reply:
we can reuse it here and in most of the other places where I have used a different variable name instead of err, but those were the reasons I didn't do it